### PR TITLE
chore(disclosure): update styles, add tests, promoto to beta

### PR DESCRIPTION
**Pull request by richbachman**
- [x] Updated Disclosure styles
- [x] File refactor
- [x] Added tests for utils functions
- [x] Promoted to beta
- [x] Updated docs anatomy tables

### CODE REVIEW

File: packages/paste-core/components/disclosure/src/Disclosure.tsx [hunk 1]
```diff
@@ -1,37 +1,13 @@
...
-
-export type {DisclosurePrimitiveInitialState as DisclosureInitialState};
-
-export interface DisclosureProps extends DisclosurePrimitiveInitialState {
-  children: NonNullable<React.ReactNode>;
-  state?: DisclosureStateReturn;
-  variant?: Variants;
-}
+import {DisclosureContext} from './DisclosureContext';
+import {DisclosureProps, DisclosurePropTypes} from './types';
```
Comment (TheSisb) [inline packages/paste-core/components/disclosure/src/Disclosure.tsx:6]: `import type`

File: packages/paste-core/components/disclosure/src/DisclosureContent.tsx [hunk 2]
```diff
@@ -1,39 +1,47 @@
 import * as React from 'react';
-import * as PropTypes from 'prop-types';
-import {DisclosurePrimitiveContent, DisclosurePrimitiveContentProps} from '@twilio-paste/disclosure-primitive';
-import {Box, BoxStyleProps} from '@twilio-paste/box';
-import {DisclosureContext, Variants} from './Disclosure';
+import {useSpring, animated} from '@twilio-paste/animation-library';
+import {Box, safelySpreadBoxProps} from '@twilio-paste/box';
+import {DisclosurePrimitiveContent} from '@twilio-paste/disclosure-primitive';
+import {DisclosureContext} from './DisclosureContext';
+import {DisclosureContentProps, DisclosureContentPropTypes} from './types';
```
Comment (TheSisb) [inline packages/paste-core/components/disclosure/src/DisclosureContent.tsx:6]: `import type`

File: packages/paste-core/components/disclosure/src/types.ts [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+  customDisabled?: boolean;
+  customFocusable?: boolean;
+  disclosureVariant: Variants;
+}
+
+export interface DisclosureContentProps extends Omit<DisclosurePrimitiveContentProps, keyof BoxStyleProps> {
+  children: NonNullable<React.ReactNode>;
+}
+
+export const DisclosurePropTypes = {
```
Comment (TheSisb) [inline packages/paste-core/components/disclosure/src/types.ts:49]: nit: a little cleaner to have a separate file (`proptypes.ts`) for these

File: packages/paste-core/components/disclosure/src/index.tsx [hunk 4]
```diff
@@ -1,3 +1,5 @@
 export * from './Disclosure';
 export * from './DisclosureHeading';
 export * from './DisclosureContent';
+export * from './DisclosureContext';
```
Comment (TheSisb) [inline packages/paste-core/components/disclosure/src/index.tsx:4]: Why do we need to expose this to the consumer?

File: packages/paste-core/components/disclosure/src/utils.ts [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+import {BoxStyleProps} from '@twilio-paste/box';
+import {HeadingProps} from '@twilio-paste/heading';
+import {ChevronDisclosureExpandedIconProps} from '@twilio-paste/icons/esm/ChevronDisclosureExpandedIcon';
+
+/**
+ * Returns the icon size based on the heading variant
+ * @param {HeadingProps['variant']} variant
+ * @returns {ChevronDisclosureExpandedIconProps} sizeIcon token for each heading variant
+ */
+export const getIconSize = (variant: HeadingProps['variant']): ChevronDisclosureExpandedIconProps['size'] => {
```
Comment (TheSisb) [inline packages/paste-core/components/disclosure/src/utils.ts:12]: Can convert this switch to a mapping:
```
{
  'heading10': 'sizeIcon90',
  ...
}
```
And if key not found default to sizeIcon20

File: packages/paste-core/components/disclosure/src/utils.ts [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+ */
+export const getIconHoverStyles = (
+  isHovered: boolean,
+  variant: HeadingProps['variant'],
+  ariaExpanded: boolean | 'true' | 'false' | undefined,
+  ariaDisabled: boolean | 'true' | 'false' | undefined,
+  space: GenericThemeShape['space']
+): Pick<BoxStyleProps, 'color' | 'transform' | 'transition'> => {
+  if (!ariaDisabled) {
+    switch (variant) {
```
Comment (TheSisb) [inline packages/paste-core/components/disclosure/src/utils.ts:47]: Can be simplified to:

```js
let slideDistance = space.space20;
if (variant === 'heading40' || variant === 'heading50' || variant === 'heading60') {
  slideDistance = space.space10;
}
return {
  color: isHovered ? 'colorText' : 'colorTextIcon',
  transform: `${isHovered ? `translateX(${slideDistance})` : `translateX(0)`} ${
    ariaExpanded ? 'rotate(0deg)' : 'rotate(-90deg)'
  }`,
  transition: 'transform 200ms ease-out',
};
```

File: packages/paste-core/components/disclosure/src/utils.ts [hunk 7]
```diff
@@ -0,0 +1,97 @@
...
+        };
+    }
+  }
+  return {
+    color: 'colorTextIcon',
+  };
+};
+
+/*
+ TODO: Should we move this hook over to a Paste hook library?
```
Comment (TheSisb) [inline packages/paste-core/components/disclosure/src/utils.ts:77]: Up to you

File: packages/paste-core/components/disclosure/src/utils.ts [hunk 8]
```diff
@@ -0,0 +1,97 @@
...
+      node.addEventListener('mouseover', handleMouseOver);
+      node.addEventListener('mouseout', handleMouseOut);
+      return () => {
+        node.removeEventListener('mouseover', handleMouseOver);
+        node.removeEventListener('mouseout', handleMouseOut);
+      };
+    }
+  }, [elementRef]);
+
+  return !!value;
```
Comment (TheSisb) [inline packages/paste-core/components/disclosure/src/utils.ts:96]: Don't need the `!!` afaict

File: packages/paste-core/components/disclosure/src/Disclosure.tsx (hunk 1 above)
Comment (SiTaggart) [reply packages/paste-core/components/disclosure/src/Disclosure.tsx:6]: We should lint this by enabling https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/consistent-type-imports.md

File: packages/paste-core/components/disclosure/src/utils.ts (hunk 6 above)
Comment (richbachman) [reply packages/paste-core/components/disclosure/src/utils.ts:47]: Good suggestion. That switch statement was a little hard to follow.

File: packages/paste-core/components/disclosure/src/index.tsx (hunk 4 above)
Comment (richbachman) [reply packages/paste-core/components/disclosure/src/index.tsx:4]: We don't. I'll remove.

File: packages/paste-core/components/disclosure/src/utils.ts (hunk 5 above)
Comment (richbachman) [reply packages/paste-core/components/disclosure/src/utils.ts:12]: Good suggestion, and more performant.

File: packages/paste-core/components/disclosure/src/Disclosure.tsx (hunk 1 above)
Comment (richbachman) [reply packages/paste-core/components/disclosure/src/Disclosure.tsx:6]: I've added the rule to our eslint config. I set it to warn for now, and I'll create a ticket to do a pass of all the components and fix the imports.

File: packages/paste-core/components/disclosure/src/utils.ts (hunk 7 above)
Comment (richbachman) [reply packages/paste-core/components/disclosure/src/utils.ts:77]: I leave it where it is for now. We can discuss adding a hooks library during focus time.

**Review by TheSisb — APPROVED** [review]
Very nicely done!
